RELEASE-NOTES.md

```diff
@@ -16,6 +16,9 @@ The Axom project release numbers follow [Semantic Versioning](http://semver.org/
 ### Deprecated
 
 ### Changed
+- Modified the API of Axom's memory management routines to not leak usage of Umpire. Instead of 
+  passing an `umpire::Allocator` object to specify an allocator, we now use the corresponding
+  integer ID associated with the allocator.
```

Comment:
Should Axom also have utility functions that specify the string ID of an allocator?

Reply:
I would prefer string rather than integer ID. It's easier to diagnose when things go wrong, since you can look at the string and have some idea of what was expected.

Reply:
Good point @davidbeckingsale.  I think we'd still also want an integer since that is how several of our users use umpire.

Reply:
There is a way to get the string name of an allocator from its integer ID and vice versa, correct?

Reply:
Yes there is.

Reply:
👍

Reply:
I'd prefer to keep the interface with just the integer IDs, at least for this PR. We can always add more functions later when these are actually needed.